Realm/RLMRealm_Private.h

```diff
@@ -69,6 +69,44 @@ FOUNDATION_EXTERN NSData *RLMRealmValidatedEncryptionKey(NSData *key);
                        schema:(RLMSchema *)customSchema
                         error:(NSError **)outError;
 
+/**
+ This method is useful only in specialized circumstances, for example, when opening Realm files
+ retrieved externally that contain a different schema than defined in your application.
+ If you are simply building an app on Realm you should consider using:
+ [defaultRealm]([RLMRealm defaultRealm]) or [realmWithPath:]([RLMRealm realmWithPath:])
+ 
+ Obtains an `RLMRealm` instance with persistence to a specific file path with
+ options.
+ 
+ @warning This method is useful only in specialized circumstances.
+ 
+ @param path                    Path to the file you want the data saved in.
+ @param key                     64-byte key to use to encrypt the data.
+ @param readonly                `BOOL` indicating if this Realm is read-only (must use for read-only files)
+ @param inMemory                `BOOL` indicating if this Realm is in-memory
+ @param dynamic                 `BOOL` indicating if this Realm is dynamic
+ @param customSchema            `RLMSchema` object representing the schema for the Realm
+ @param disableFormatUpgrade    Whether Realm may upgrade this files format version if required
+ @param outError                If an error occurs, upon return contains an `NSError` object
+ that describes the problem. If you are not interested in
+ possible errors, pass in NULL.
+ 
+ @return An `RLMRealm` instance.
+ 
+ @see RLMRealm defaultRealm
+ @see RLMRealm realmWithPath:
+ @see RLMRealm realmWithPath:readOnly:error:
+ @see RLMRealm realmWithPath:encryptionKey:readOnly:error:
+ */
++ (instancetype)realmWithPath:(NSString *)path
```

Comment:
The other similar methods predate `RLMRealmConfiguration`. Now that it exists I don't think there's any reason to have this in Realm proper, when it can easily be a helper function in the browser code that uses it.